catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTableOperations.java

```diff
@@ -25,34 +73,515 @@ protected String generateCreateTableSql(
+        Arrays.stream(indexes)
+            .map(index -> String.format("INDEX %s (%s)", index.name(), index.fieldNames()[0][0]))
+            .collect(Collectors.joining(",\n"));
+
+    sqlBuilder.append(",").append(NEW_LINE).append(indexSql);
+  }
+
+  @Override
+  protected boolean getAutoIncrementInfo(ResultSet resultSet) throws SQLException {
+    return "YES".equalsIgnoreCase(resultSet.getString("IS_AUTOINCREMENT"));
+  }
+
+  @Override
+  protected Map<String, String> getTableProperties(Connection connection, String tableName)
+      throws SQLException {
+
+    String showCreateTableSQL = String.format("SHOW CREATE TABLE `%s`", tableName);
+
+    StringBuilder createTableSqlSb = new StringBuilder();
+    try (Statement statement = connection.createStatement();
+        ResultSet resultSet = statement.executeQuery(showCreateTableSQL)) {
+      while (resultSet.next()) {
+        createTableSqlSb.append(resultSet.getString("Create Table"));
+      }
+    }
+
+    String createTableSql = createTableSqlSb.toString();
+
+    if (StringUtils.isEmpty(createTableSql)) {
+      throw new NoSuchTableException(
+          "Table %s does not exist in %s.", tableName, connection.getCatalog());
+    }
+
+    return Collections.unmodifiableMap(DorisUtils.extractPropertiesFromSql(createTableSql));
+  }
+
+  @Override
+  protected List<Index> getIndexes(Connection connection, String databaseName, String tableName)
+      throws SQLException {
+    String sql = String.format("SHOW INDEX FROM `%s` FROM `%s`", tableName, databaseName);
+
+    // get Indexes from SQL
+    try (PreparedStatement preparedStatement = connection.prepareStatement(sql);
+        ResultSet resultSet = preparedStatement.executeQuery()) {
+
+      List<Index> indexes = new ArrayList<>();
+      while (resultSet.next()) {
+        String indexName = resultSet.getString("Key_name");
+        String columnName = resultSet.getString("Column_name");
+        indexes.add(
+            Indexes.of(Index.IndexType.PRIMARY_KEY, indexName, new String[][] {{columnName}}));
+      }
+      return indexes;
+    } catch (SQLException e) {
+      throw exceptionMapper.toGravitinoException(e);
+    }
+  }
+
+  @Override
+  protected void correctJdbcTableFields(
+      Connection connection, String databaseName, String tableName, JdbcTable.Builder tableBuilder)
+      throws SQLException {
+
+    if (StringUtils.isNotEmpty(tableBuilder.comment())) {
+      return;
+    }
+
+    // Doris Cannot get comment from JDBC 8.x, so we need to get comment from sql
+    String comment = "";
+    String sql =
+        "SELECT TABLE_COMMENT FROM information_schema.TABLES WHERE TABLE_SCHEMA = ? AND TABLE_NAME = ?";
+    try (PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+      preparedStatement.setString(1, databaseName);
+      preparedStatement.setString(2, tableName);
+
+      try (ResultSet resultSet = preparedStatement.executeQuery()) {
+        while (resultSet.next()) {
```

Comment:
is it possible to retrive multi comments?

Reply:
Mostly resultSet just returns one line, but I think it's necessary to iterate through the resultSet.

Reply:
This seems a bug if geting more than one line?

Reply:
I can't temporarily construct a multi-line test case, but using a while loop here to ensure all resultSets are read seems to pose no problem.